### PR TITLE
issue #28 Rename to extensions

### PR DESCRIPTION
**Pull request by mloufra**
Signed-off-by: mloufra <mloufra@amazon.com>

### Description
Rename method names, class names, variable names from plugin to extensions in the sdk codebase and related OpenSearch code

Equivalent PR on SDK: https://github.com/opensearch-project/opensearch-sdk/pull/72
 
### Issues Resolved
https://github.com/opensearch-project/opensearch-sdk/issues/28
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ryanbogan — APPROVED** [review]

File: server/src/main/java/org/opensearch/discovery/InitializeExtensionsRequest.java [hunk 1]
```diff
@@ -19,23 +19,23 @@
 import java.util.Objects;
 
 /**
- * PluginRequest to intialize plugin
+ * InitializeExtensionsRequest to intialize plugin
```
Comment (owaiskazi19) [inline server/src/main/java/org/opensearch/discovery/InitializeExtensionsRequest.java:22]:
```suggestion
 * InitializeExtensionsRequest to initialize extension
```

File: server/src/main/java/org/opensearch/discovery/InitializeExtensionsResponse.java [hunk 2]
```diff
@@ -40,18 +40,18 @@
 import java.util.Objects;
 
 /**
- * PluginResponse to intialize plugin
+ * InitializeExtensionsResponse to intialize plugin
```
Comment (owaiskazi19) [inline server/src/main/java/org/opensearch/discovery/InitializeExtensionsResponse.java:43]:
```suggestion
 * InitializeExtensionsResponse to initialize extension
```

File: server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java [hunk 3]
```diff
@@ -229,15 +229,16 @@ public void extensionsInitialize() {
 
     private void extensionInitialize(DiscoveryNode extensionNode) {
         final CountDownLatch inProgressLatch = new CountDownLatch(1);
-        final TransportResponseHandler<PluginResponse> pluginResponseHandler = new TransportResponseHandler<PluginResponse>() {
+        final TransportResponseHandler<InitializeExtensionsResponse> pluginResponseHandler = new TransportResponseHandler<
```
Comment (owaiskazi19) [inline server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java:232]: `pluginResponseHandler` -> `extensionResponseHandler`

**Review by owaiskazi19 — APPROVED** [review]
LGTM

**Pull request merged**